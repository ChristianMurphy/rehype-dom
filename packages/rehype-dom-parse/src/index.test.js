import unified from 'unified';

import parse from './index';

describe('rehype-dom-parse', () => {
  const processor = unified().use(parse).freeze();

  it('should parse a complete document', () => {
    const result = processor()
      .data('settings', { fragment: false })
      .parse('<title>Hi</title><h2>Hello world!');
    expect(result).toMatchSnapshot();
  });

  it('should parse a fragment', () => {
    const result = processor()
      .data('settings', { fragment: true })
      .parse('<title>Hi</title><h2>Hello world!');
    expect(result).toMatchSnapshot();
  });
});
