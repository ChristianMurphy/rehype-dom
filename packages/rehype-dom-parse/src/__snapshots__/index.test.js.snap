// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehype-dom-parse should parse a complete document 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Hi",
                },
              ],
              "properties": Object {},
              "tagName": "title",
              "type": "element",
            },
          ],
          "properties": Object {},
          "tagName": "head",
          "type": "element",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Hello world!",
                },
              ],
              "properties": Object {},
              "tagName": "h2",
              "type": "element",
            },
          ],
          "properties": Object {},
          "tagName": "body",
          "type": "element",
        },
      ],
      "properties": Object {},
      "tagName": "html",
      "type": "element",
    },
  ],
  "type": "root",
}
`;

exports[`rehype-dom-parse should parse a fragment 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Hi",
        },
      ],
      "properties": Object {},
      "tagName": "title",
      "type": "element",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Hello world!",
        },
      ],
      "properties": Object {},
      "tagName": "h2",
      "type": "element",
    },
  ],
  "type": "root",
}
`;
